include/range/v3/begin_end.hpp

```diff
@@ -103,8 +105,10 @@ namespace ranges
             public:
                 template<typename Rng>
                 constexpr auto operator()(Rng && rng) const
-                    noexcept(noexcept(begin_fn::impl(static_cast<Rng &&>(rng), 0))) ->
-                    detail::decay_t<decltype(begin_fn::impl(static_cast<Rng &&>(rng), 0))>
+#if !defined(__GNUC__) || __GNUC__ >= 5
+                    noexcept(noexcept(begin_fn::impl(static_cast<Rng &&>(rng), 0)))
+#endif
+                    -> detail::decay_t<decltype(begin_fn::impl(static_cast<Rng &&>(rng), 0))>
```

Comment:
I strongly agree. It seems the problem only occurs if the range is any_view, but I have not been able to provide a better solution.

It was more for the Travis to check whether it passes the tests since I did not have a machine with gcc4. It would actually pass the tests I think, since Travis failed with unrelated apt-get error on two clang tests before even trying to build.